bears/c_languages/CSecurityBear.py

```diff
@@ -0,0 +1,30 @@
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+
+@linter(executable="flawfinder",
+        output_format="regex",
+        output_regex=r'.+:(?P<line>\d+):(?P<column>\d+):\s*'
+                     r'\[(?P<severity>\d)\]\s*'
+                     r'\((?P<origin>.+)\) (?P<message>.+)',
+        severity_map={"1": RESULT_SEVERITY.INFO,
+                      "2": RESULT_SEVERITY.INFO,
+                      "3": RESULT_SEVERITY.NORMAL,
+                      "4": RESULT_SEVERITY.NORMAL,
+                      "5": RESULT_SEVERITY.MAJOR})
```

Comment:
done :)


2016-07-06 16:27 GMT+01:00 Zatreanu Adrian-Gabriel <notifications@github.com
>:

> In bears/c_languages/CSecurityBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/566#discussion_r69752900>
> :
>
> > @@ -0,0 +1,30 @@
> > +from coalib.bearlib.abstractions.Linter import linter
> > +from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
> > +
> > +
> > +@linter(executable="flawfinder",
> > +        output_format="regex",
> > +        output_regex=r'.+:(?P<line>\d+):(?P<column>\d+):\s*'
> > +                     r'\[(?P<severity>\d)\]\s*'
> > +                     r'\((?P<origin>.+)\) (?P<message>.+)',
> > +        severity_map={"1": RESULT_SEVERITY.INFO,
> > +                      "2": RESULT_SEVERITY.INFO,
> > +                      "3": RESULT_SEVERITY.NORMAL,
> > +                      "4": RESULT_SEVERITY.NORMAL,
> > +                      "5": RESULT_SEVERITY.MAJOR})
>
> hm maybe you could add a prerequisite check and fail message, such as
> CSSAutoPrefixBear has for example? :) wouldn't hurt. you could mention in
> that message that you need to run flawfinder with python2
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/566/files/b02a9bba4e6136f442236cd76e25100035d4bb0c#r69752900>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xogcOHs3vVE2mKPPw1O6O2i9z0Nbwks5qS8lKgaJpZM4JF479>
> .
>